examples/error-type-binding-pattern/error_type_binding_pattern.bal

```diff
@@ -1,37 +1,38 @@
 import ballerina/io;
 
-type SampleError error<string, map<anydata|error>>;
+type SampleErrorData record {
+    string message?;
+    error cause?;
+    string info;
+    boolean fatal;
+};
+
+type SampleError error<string, SampleErrorData>;
 
 public function main() {
     // This error-type binding pattern will destructure an `error` of the type `SampleError`, and create two variables as follows:
     // The value of the reason string in the`SampleError` will be set to a new `string` variable `reason`.
-    // The value of the detail mapping will be set to a new `map<anydata|error>` variable `detail`.
-    SampleError error(reason, detail) = getSampleError();
+    // The values in the detail mapping will be set to new variables `info` and `fatal`.
```

Comment:
```suggestion
    // The values in the detailed mapping will be set to the new variables `info` and `fatal`.
```

Reply:
Error object has two fields called "reason" and "detail". The "detail" result (mapping) is stated here.